lib/Basic/Remangler.cpp

```diff
@@ -679,6 +679,24 @@ void Remangler::mangleDependentGenericSameTypeRequirement(Node *node) {
   mangleDependentGenericParamIndex(NumMembersAndParamIdx.second);
 }
 
+void Remangler::mangleDependentGenericLayoutRequirement(Node *node) {
+  auto NumMembersAndParamIdx = mangleConstrainedType(node->getChild(0).get());
+  switch (NumMembersAndParamIdx.first) {
+    case -1: Buffer << "RL"; return; // substitution
+    case 0: Buffer << "Rl"; break;
+    case 1: Buffer << "Rm"; break;
+    default: Buffer << "RM"; break;
+  }
+  mangleDependentGenericParamIndex(NumMembersAndParamIdx.second);
+  assert(node->getChild(1)->getKind() == Node::Kind::Identifier);
+  assert(node->getChild(1)->getText().size() == 1);
+  Buffer << node->getChild(1)->getText()[0];
+  if (node->getNumChildren() >=3)
+    mangleChildNode(node, 2);
```

Comment:
The children (which represent the RHS of the requirement, i.e. the layout constraint itself) are: the letter for the kind of layout constraint and one or two numbers for size/alignment. No types. 

The only type in play is the LHS of the requirement. This is the dependent generic param index. So, IMHO, everything is OK.